src/formats/tga.rs

```diff
@@ -0,0 +1,40 @@
+use std::io::{BufRead, Seek, SeekFrom};
+
+use crate::{ImageResult, ImageSize};
+
+pub fn size<R: BufRead + Seek>(reader: &mut R) -> ImageResult<ImageSize> {
+    reader.seek(SeekFrom::Start(0))?;
+
+    let mut header = [0; 18];
+    reader.read_exact(&mut header)?;
+
+    let width = u16::from_le_bytes([header[12], header[13]]) as usize;
+    let height = u16::from_le_bytes([header[14], header[15]]) as usize;
+
+    Ok(ImageSize { width, height })
+}
+
+pub fn matches(header: &[u8]) -> bool {
```

Comment:
So this is the most tricky part since I really want to avoid false positives.

Using [Paul Bourke's TGA format specification](http://www.paulbourke.net/dataformats/tga/) I think there's a better way to verify if a header is TGA. 

For example, lets take `image_type` 9:

Current implementation:
- `image_type` is a valid number (9)
- `colormap_type` of 0 and 1 are both valid

This means any file with the 3rd byte being 9 and the 2nd byte being 0 or 1 will be a TGA file.

My proposed extension of this would be using the information at the bottom of the page I linked:

- `image_type` is 9 which is an expected value
- With `image_type` 9, `colormap_type` must be 1 so 0 would be invalid/not TGA
- Byte 8 (Color Map Entry Size) must be one of 16/24/32
- Byte 18 (Image Descriptor Byte) bits 0-3 should be 0/1 if byte 8 is 16, 0 if byte 8 is 24, and 8 if byte 8 is 32. Bit 4 should also be 0.

I think this sort of cross referencing based on `image_type` would give a lot more confidence in an image being TGA or not. Unfortunately Paul's reference only has examples for types 1, 2, 9, and 10. I haven't really done digging to see if there are examples for 0, 3, 11, 32, or 33 yet.

If references for those are hard to come by I would prefer if this starts support with 1, 2, 9, and 10 with a comment about the other values and maybe when references are found we can add them.

Reply:
By the way this is something I might want to benchmark, but if doing a seek to the end of the file and reading the TGA 2.0 footer doesn't take too long that is also an option that is much more robust. This does not handle earlier than 2.0 TGA files though.

If you want to try this out do something like:
```rust
reader.seek(SeekFrom::End(18));
// Check that next 16 bytes are TRUEVISION-XFILE
// Optionally also check that byte 17 is . and byte 18 is null
```

I'm not sure what happens if the reader is less than 18 bytes though.